### PR TITLE
Add function whose return data from .tcx files

### PR DESCRIPTION
**Pull request by nero2310**
This function return data as Datatable object list, aslo return
Dictonary with exceptions

### CODE REVIEW

File: hdp/process/load.py [hunk 1]
```diff
@@ -183,9 +186,36 @@ def load_gpx(gpx_files: list, params: dict):
...
+        except Exception as e:
+            exceptions_dict[name] = str(e)
+        else:
+            data_table.name = name
+            data_table_list.append(data_table)
+    return data_table_list, exceptions_dict
 
 
 # Team 4
 def load_jpg(jpg_files: list, params: dict):
```
Comment (mdyzma) [inline hdp/process/load.py]: This function ddoes not do anytjhing Use YAGNI (you are not gonna need it). Please, do not place code without function.

I suggest use Project tab and create issue/bug.

File: hdp/process/load.py [hunk 2]
```diff
@@ -183,9 +186,36 @@ def load_gpx(gpx_files: list, params: dict):
...
+            for trackpoint in activity.find_all("trackpoint"):
+                point = {element.name: element.text for element in trackpoint.children if
+                         isinstance(element, Tag) and len(element.findChildren()) == 0}
+                for element in trackpoint.children:
+                    if isinstance(element,Tag) and len(element.findChildren())>1:
+                        for value in element.children:
+                            if isinstance(value,NavigableString):
+                                point[value.name] = value.string
+                            elif isinstance(value,Tag):
+                                point[value.name] = value.text.strip()
```
Comment (mdyzma) [inline hdp/process/load.py]: Too complicated/nested function, try to extract to higher level function.

i.e.:

`def extract_path():
    pass`


```python
def extract_all_trackpoints(node_name="trackpoint"):
    for trackpoint in activity.find_all(node_name):
        points = {element.name: element.text for element in trackpoint.children if
             isinstance(element, Tag) and len(element.findChildren()) == 0}
    return points

def extract_points(trackpoints):
    point = {}
    for element in trackpoints.children:
        if isinstance(element,Tag) and len(element.findChildren())>1:
            for value in element.children:
                if isinstance(value,NavigableString):
                    point[value.name] = value.string
                elif isinstance(value,Tag):
                    point[value.name] = value.text.strip()
    return point
```

If it still complains refactor further.

File: hdp/process/load.py [hunk 3]
```diff
@@ -1,4 +1,7 @@
 import pandas as pd
+from bs4 import BeautifulSoup
+from bs4 import NavigableString, Tag
+
```
Comment (mdyzma) [inline hdp/process/load.py:5]: Import proper typing suggestions
```suggestion
from typing import List, Tuple, Dict
```

File: hdp/process/load.py [hunk 4]
```diff
@@ -183,9 +186,36 @@ def load_gpx(gpx_files: list, params: dict):
 
 # Team 3
 def load_tcx(tcx_files: list, params: dict):
```
Comment (mdyzma) [inline hdp/process/load.py]:
```suggestion
def load_tcx(tcx_files: List[str], params: Dict) -> Tuple:
```

**Review by mdyzma — CHANGES_REQUESTED** [review]

File: hdp/process/load.py (hunk 2 above)
Comment (nero2310) [reply hdp/process/load.py]: Should I define auxiliary functions inside load_tcx or outside?

**Comment by mdyzma** [comment]
Outside

czw., 2 lip 2020, 16:30 użytkownik nero2310 <notifications@github.com>
napisał:

File: hdp/process/load.py (hunk 1 above)
Comment (nero2310) [reply hdp/process/load.py]: Do you mean this load_jpg function? I am pretty sure this is empty for a purpose.

**Pull request merged**